src/Sarif.UnitTests/FileDiffingTests.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the MIT        
+// license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using Microsoft.CodeAnalysis.Sarif.Readers;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.UnitTests
+{
+    public class FileDiffingTests
+    {
+        protected string GenerateDiffCommand(string expected, string actual)
+        {
+            expected = Path.GetFullPath(expected).Replace(@"\", @"\\");
+            actual = Path.GetFullPath(actual).Replace(@"\", @"\\");
+
+            string beyondCompare = TryFindBeyondCompare();
+            if (beyondCompare != null)
+            {
+                return String.Format(CultureInfo.InvariantCulture, "\"{0}\" \"{1}\" \"{2}\" /title1=Expected /title2=Actual", beyondCompare, expected, actual);
+            }
+
+            return String.Format(CultureInfo.InvariantCulture, "windiff \"{0}\" \"{1}\"", expected, actual);
```

Comment:
>windiff [](start = 64, length = 7)

It's worse than that. These tests won't run at all on a non-Windows machine.

Reply:
And ok, ok, it's not that the tests won't run on non-Windows, it's that the message will be wrong.

---
In reply to: [205981167](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205981167) [](ancestors = 205981167)

Reply:
Right, part of this test methodology is to assist the developer into jumping into a diffing experience in as efficient a way as possible. Currently, we don't emit these diff commands on the console when testing (a gap). also, you have to click through to the 'Output' pane in the IDE, another gap. Ideally, you get a command you can scrape easily for bulk comparison of all diffs (with this sort of data-driven testing approach).

---
In reply to: [205982688](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205982688) [](ancestors = 205982688,205981167)